### DISCUSSION TITLE
STOP before it's too late

### DISCUSSION THREAD

**Discussion by Juka117** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#539] [General]
Hi Melvin,

You have taken on a great great initiative. I will honestly pay good money to get this supported.
However good developers here, yourself included for sure, should know that the way the project is laid out, will not survive another year max. You can't have 1 main JS file with 5000+ lines and we have to accept this. Even progress is suffering because of this.

I strongly strongly suggest you stop new development, even if it takes you another month, and kindly divide/organize the code so that we can help out more and you develop faster yourself too. I was one of the early forkers who literally gave up after my first pull request. Now it has become crazy complex, almost impossible to follow.

PS: if I'm doing something wrong, please correct me but I see 1 file with 5.5k lines of code.

Thanks,
Justin

**Comment by sumeshir26** [comment]
@Juka1186 @Melvin-Abraham mentioned [here ](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/issues/477#issuecomment-892881080) that he will rerwite the codebase in react native once the next release is out
